### PR TITLE
Change restrictor for Orgaadmin

### LINKED ISSUE #4380: Restriction change for Orgaadmin
The available options for the orga admin should be changed 
The autoupdate already changed this here: https://github.com/OpenSlides/openslides-autoupdate-service/pull/1053

Short version of the wanted behaviour: The Orgaadmin should have the same possibilities as the superadmin (minus explizit superadmin options)

Places which should change: 
- Orgaadmin should always be able to create meeting (even if they are not a committeeadmin)
- An Orgaadmin should be allowed to enter a meeting and see all buttons/ information as the superadmin
- The guards need allow the Orgaadmin to navigate to the meeting
- If a meeting is **closed from inside**: the visible contents for a orgaadmin should be as restricted as they are for a superadmin

### PR DESCRIPTION
**Pull request by Elblinator**
closes #4380 

There are still some parts which were not changed and are still not presented by the Orgaadmin
@bspekker, @emanuelschuetze, @MSoeb Please state which of these options are wanted for the Orgaadmin:
A) Saml options are not displayed for the Orgaadmin
B) Impressum "Check datastore" and "Thoroughly check datastore (unsafe)"
C) Meeting edit > Settings for Jitsi domain 
D) Committee list > meeting import
E) Meeting export

This PR is allowing an orgaadmin to see every data, option etc. if they should be able to see it. To give an Orga admin the possibility to use all action the backend needs to be changed as well

### CODE REVIEW

File: client/src/app/site/services/operator.service.ts [hunk 1]
```diff
@@ -659,18 +659,9 @@ export class OperatorService {
...
-        if (this.isOrgaManager) {
-            return true;
-        }
-        return this.hasCommitteePermissionsNonAdminCheck(committeeId, ...permissionsToCheck);
-    }
-
-    public hasCommitteePermissionsNonAdminCheck(committeeId: Id | null, ...permissionsToCheck: CML[]): boolean {
-        // A superadmin can still do everything
-        if (this.isSuperAdmin) {
+    public hasCommitteePermissionsOrOrgaPermissions(committeeId: Id | null, ...permissionsToCheck: CML[]): boolean {
```
Comment (bastianjoel) [inline client/src/app/site/services/operator.service.ts:662]: Why rename this? I think the old name is still sufficient and not changing it leads to less changes.

File: client/src/app/ui/directives/perms/cml-perms.directive.spec.ts [hunk 2]
```diff
@@ -63,12 +60,8 @@ class MockOperatorService {
         return checkPerms.some(perm => perm === this._oml);
     }
 
-    public hasCommitteePermissions(committeeId: Id | null, ...checkPerms: CML[]): boolean {
-        return this._isAdmin || this.hasCommitteePermissionsNonAdminCheck(committeeId, ...checkPerms);
-    }
-
-    public hasCommitteePermissionsNonAdminCheck(committeeId: Id | null, ...checkPerms: CML[]): boolean {
-        return checkPerms.some(perm => this._permList.includes(perm));
+    public hasCommitteePermissionsOrOrgaPermissions(committeeId: Id | null, ...checkPerms: CML[]): boolean {
```
Comment (bastianjoel) [inline client/src/app/ui/directives/perms/cml-perms.directive.spec.ts:63]:
```suggestion
    public hasCommitteePermissions(committeeId: Id | null, ...checkPerms: CML[]): boolean {
```

File: client/src/app/site/services/operator.service.ts [hunk 3]
```diff
@@ -588,7 +588,7 @@ export class OperatorService {
             // console.warn(`has perms: Usage outside of meeting!`);
             return false;
         }
-        if (this.isSuperAdmin && !this.activeMeeting.locked_from_inside) {
+        if ((this.isSuperAdmin || this.isOrgaManager) && !this.activeMeeting.locked_from_inside) {
```
Comment (bastianjoel) [inline client/src/app/site/services/operator.service.ts:591]: This check is done quite often. Maybe add an property. Note that you are doing this check also outside of the operator service. So maybe it even makes sense to add that property as public. 

```suggestion
        if (this.canSkipPermissionCheck && !this.activeMeeting.locked_from_inside) {
```

File: client/src/app/site/pages/organization/pages/committees/pages/committee-detail/modules/committee-detail-view/components/committee-detail-view/committee-detail-view.component.ts [hunk 4]
```diff
@@ -30,12 +30,8 @@ export class CommitteeDetailViewComponent extends BaseUiComponent {
     public forwardingExpanded = false;
     public requireDuplicateFrom = false;
 
-    public get canManageMeetingsInCommittee(): boolean {
-        return this.operator.hasCommitteePermissionsNonAdminCheck(this.committeeId, CML.can_manage);
-    }
-
-    public get canManageCommittee(): boolean {
-        return this.operator.hasCommitteePermissions(this.committeeId, CML.can_manage);
+    public get canManageCommitteeOrMeetingsInCommittee(): boolean {
```
Comment (bastianjoel) [inline client/src/app/site/pages/organization/pages/committees/pages/committee-detail/modules/committee-detail-view/components/committee-detail-view/committee-detail-view.component.ts:33]: This is confusing. You are delegating this to a lower level so keep the name simple. 
```suggestion
    public get canManageCommittee(): boolean {
```

**Review by bastianjoel — CHANGES_REQUESTED** [review]

**Comment by Elblinator** [comment]
A-E is not wanted for the Orgaadmin

An Orgadmin should not be able to delete meetings if the meeting is locked from inside and an Orgaadmin should not be able to dublicate a meeting which is locked from inside

**Comment by Elblinator** [comment]
The PR still needs to make sure that locked meetings are displayed correctly

From a Comment from the backend: 
An Orgaadmin should only be allowed to delete and dublicate a meeting if they have the "meeting.can_manage_settings" from this meeting if the meeting is locked from inside

**Comment by Elblinator** [comment]
> An Orgaadmin should only be allowed to delete and dublicate a meeting if they have the "meeting.can_manage_settings" from this meeting if the meeting is locked from inside

we (@bastianjoel and @Elblinator)  talked about this requirement. 
What I decided to implement in the end: 
A user does not need the permission meeting.can_manage_settings but they need to be an admin of the group to be all0owed to dublicate/delete a meeting

Why did I change the requirment:
Loading every group at this place will create performance issues...
Loading the groups entails to load every permission from every group (and some more) from every meeting in this committee

File: client/src/app/site/pages/organization/pages/committees/modules/committee-meeting-preview/committee-meeting-preview.component.html [hunk 5]
```diff
@@ -145,7 +145,7 @@
                 <span>{{ 'Public template' | translate }}</span>
             </button>
         }
-        @if (!isLockedFromInside) {
+        @if (canEditMeetingSettings()) {
```
Comment (bastianjoel) [inline client/src/app/site/pages/organization/pages/committees/modules/committee-meeting-preview/committee-meeting-preview.component.html:148]: This should not be used directly within a template which does not use `OnPush`.

**Review by bastianjoel — CHANGES_REQUESTED** [review]

File: client/src/app/site/pages/organization/pages/committees/modules/committee-meeting-preview/committee-meeting-preview.component.ts [hunk 6]
```diff
@@ -64,14 +65,46 @@ export class CommitteeMeetingPreviewComponent {
...
+
+    /**
+     * clear the Subscriptions
+     */
+    public ngOnDestroy(): void {
+        if (this._canEditMeetingSubscription) {
+            this._canEditMeetingSubscription.unsubscribe();
+            this._canEditMeetingSubscription = null;
+        }
+        this.cd.detach();
```
Comment (bastianjoel) [inline client/src/app/site/pages/organization/pages/committees/modules/committee-meeting-preview/committee-meeting-preview.component.ts:105]:
```suggestion
```

File: client/src/app/site/pages/organization/pages/committees/modules/committee-meeting-preview/committee-meeting-preview.component.ts [hunk 7]
```diff
@@ -64,14 +65,46 @@ export class CommitteeMeetingPreviewComponent {
...
 
+    public get canEditMeetingSetting(): boolean {
+        return this._canEditMeetingSetting;
+    }
+
+    private _canEditMeetingSetting = true;
+    private _canEditMeetingSubscription: Subscription;
+
     public constructor(
+        private cd: ChangeDetectorRef,
```
Comment (bastianjoel) [inline client/src/app/site/pages/organization/pages/committees/modules/committee-meeting-preview/committee-meeting-preview.component.ts:76]:
```suggestion
```

**Review by bastianjoel — APPROVED** [review]
Please remove the cd stuff. Rest is okay now.

**Review by MSoeb — CHANGES_REQUESTED** [review]
Found some permission issues:

- <s>Committee menu: Create a meeting is not possible. Orgaadmin needs to be committee admin to create a committee. Following the superadmin permissions, this should be possible without setting the orgaadmin as committee admin.</s> see https://github.com/OpenSlides/openslides-client/issues/4608

- Committe menu > meeting edit: Edit is not possible. Wanted behavior: Same as above. 
image

- Meeting > Participants > Add or edit a user: Adding or editing a participant doesn't work. Permission is missing. Wanted behavior: Should work, without being a part (admin) of the meeting.
image

- <s>Motions > create a motion: Didn't work if the submitter is a new participant. This is a follow up of the previous point. I assume, that this behavior will happen on every action that requires to add or edit a participant. Wanted behavior: Creating a motionwithout being part of the meeting should be possible. Just like superadmins can do it.<s/> see https://github.com/OpenSlides/openslides-client/issues/4607

**Comment by Elblinator** [comment]
Please state your oroblem with the client, thiese are all issues with the backend

**Review by MSoeb — CHANGES_REQUESTED** [review]
Found two errors, which need to be fixed.

List of speakers: 
- [ ] It is not possible to add an orgaadmin, which is not part of a meeting, to a List of speakers or create an point of order. In BOTH cases the SAME error message appears. See the error adding to list of speakers in the console and as a snackbar by creating a point of order below. I'm not sure, where I should put this error. It could be relevant for client or/and backend. So please decide where it should belong.

Error for 'Add me to' List of speakers
image

Error for 'point of order' - FYI: The network tab showed me no information about response or payload.
image

Reproduction: 
precondition: existing meeting with at least one agenda item
1. open accounts > create a new user as orgaadmin
2. login as new user > open the meeting > open list of speakers of the agenda item > try to add you to list of speakers and try to create a point of order.

----

- [ ] Editing closed meetings as orgaadmin which is not part of the closed meeting: It is currently possible to enter /change something in the edit mask. Saving it not possible, which is correct behavior. > Wanted: The edit can be opened, but all text fields should be grayed out, so that nothing ca be entered.
